Kernel/Modules.cmake

```diff
@@ -0,0 +1,162 @@
+if (NOT EXISTS "${CMAKE_SOURCE_DIR}/Build/${SERENITY_ARCH}/Kernel/config.ini")
+    configure_file("${CMAKE_SOURCE_DIR}/Kernel/config.ini.template" "${CMAKE_SOURCE_DIR}/Build/${SERENITY_ARCH}/Kernel/config.ini" COPYONLY)
+endif()
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "${CMAKE_SOURCE_DIR}/Build/${SERENITY_ARCH}/Kernel/config.ini")
+
+
+# Default values for default setup
+set(COMPILE_IDE "y")
+set(COMPILE_AHCI "y")
+set(COMPILE_RAMDISK "y")
+
+set(COMPILE_E1000 "y")
+set(COMPILE_E1000E "y")
+set(COMPILE_RTL8139 "y")
+set(COMPILE_RTL8168 "y")
+set(COMPILE_NE2000 "y")
```

Comment:
`if()` interprets a whole host of strings as "truthy" and "falsey" https://cmake.org/cmake/help/latest/command/if.html#basic-expressions

> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number. False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND. Named boolean constants are case-insensitive. 

So we could instead `set(COMPILE_FOO ON)` or `set(COMPILE_FOO TRUE)` or some such, and then put any of those strings in the .ini file. even "Y" (not 100% sure if "y" works), and then all the checks below become one of: 

```
if (COMPILE_FOO)
   // compile foo
endif()
```
or 
```
if ("${COMPILE_FOO}")
   // compile foo
endif()
```
but I think the first one should work. 

if "y" doesn't work and you want to keep it lowercase in the .ini file, then a simple `string(TOUPPER "${value}" value)` should do the trick

Reply:
That's quite nice. I'll try to make it to work then :)

Reply:
Before I apply this suggestion I want to ensure I fix the bugs I created with Clang CI so it is marked green before I change anything else. Then I'll fix this, and hopefully after that I'll figure out the ifdefs and this will be ready for being merged :)

Reply:
This seems to work out quite nicely, thank you again for the suggestion! :)

Reply:
FYI, the boolan constants are checked case insensitively, except for `NOTFOUND`.